### PR TITLE
Prevent release proposals with invalid labels

### PR DESCRIPTION
**Pull request by uurien**
### What does this PR do?

It adds a workflow that fails if a release proposal has a commit with invalid `do-not-land-on` tag.
### Motivation

Prevent releases with not compatible commits

### Examples
Error in 2.x proposal: https://github.com/DataDog/dd-trace-js/pull/2734
Error in 3.x proposal: https://github.com/DataDog/dd-trace-js/pull/2733

### Plugin Checklist

- [ ] Unit tests.
- [ ] TypeScript [definitions][1].
- [ ] TypeScript [tests][2].
- [ ] API [documentation][3].
- [ ] CircleCI [jobs/workflows][4].
- [ ] Plugin is [exported][5].

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/documentation/blob/master/content/en/tracing/trace_collection/library_config/nodejs.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### CODE REVIEW

**Review by Qard — COMMENTED** [review]
Is it possible to use branch patterns and capture the branch name instead of having separate files like this? It would be nice if we don't need to create new files all the time.

File: .github/workflows/release-proposal.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Release Proposal PR check
+
+on:
+  pull_request:
+    branches:
+      - v2.x
+      - v3.x
```
Comment (uurien) [inline .github/workflows/release-proposal.yml:7]: @Qard your are right about the number of files,updated, here and in sample PRs.
We will not have to create new file for each version, but we should add new branch in this list for each version

```suggestion
on:
  pull_request:
    branches:
      - v2.x
      - v3.x
      - v4.x
      - v5.x
      - ...
```

Comment (rochdev) [reply .github/workflows/release-proposal.yml:7]: Why not just use a pattern?

Comment (uurien) [reply .github/workflows/release-proposal.yml]: Because I didn't know that it was possible 🤦  changed

**Review by Qard — APPROVED** [review]

**Pull request merged**